vault: Support multiple config files (no secrets in store)

roberth opened the pull request:
###### Motivation for this change

Configure storageBackend securely.
Although vault guarantees confidentiality and integrity when its backend is compromised, a leak still compromises the storage backend itself and the availability of vault.

Closes #107323 (earlier attempt, wrong approach)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc maintainers @LnL7 @rushmorem @offlinehacker @pradeepchhetri 
cc @aanderse

roberth committed:
nixosTests.vault: increase memorySize to 512

roberth committed:
nixos/vault: Allow multiple config files

roberth committed:
nixosTests.vault-postgresql: init

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/438

roberth commented:
@LnL7 @rushmorem @offlinehacker @pradeepchhetri  Could you please review?

roberth committed:
nixos/vault: extraConfigPaths -> extraSettingsPaths

Align with RFC42 language, even if in advance of the actual settings
attribute.

cpcloud reviewed (approved):
This LGTM!

roberth commented:
I'm not expecting anything from the listed package maintainers (@LnL7 @rushmorem @offlinehacker @pradeepchhetri) after 34 days of silence, also counting the previous iteration of this PR.
